op-service/txmgr/txmgr.go

```diff
@@ -403,6 +394,34 @@ func (m *SimpleTxManager) publishAndWaitForTx(ctx context.Context, tx *types.Tra
 	}
 }
 
+// sendWithUnderpricedRetry sends the transaction to the transaction pool. If the transaction is
+// underpriced, it will bump the gas price and retry.
+func (m *SimpleTxManager) sendWithUnderpricedRetry(ctx context.Context, tx *types.Transaction, l log.Logger) error {
+	for {
+		err := m.backend.SendTransaction(ctx, tx)
+		if err == nil {
+			return nil
+		}
+
+		switch {
+		case errStringMatch(err, txpool.ErrReplaceUnderpriced):
+			l.Warn("transaction replacement is underpriced", "err", err)
+			m.metr.TxPublished("tx_replacement_underpriced")
+		case errStringMatch(err, txpool.ErrUnderpriced):
+			l.Warn("transaction is underpriced", "err", err)
+			m.metr.TxPublished("tx_underpriced")
+		default:
+			return err
+		}
+
+		// Increase the gas price & retry
+		tx, err = m.increaseGasPrice(ctx, tx)
```

Comment:
this modified tx needs to be returned for the `waitMined` to work, will refactor